doc/api/stream.md

```diff
@@ -1859,6 +1859,46 @@ run().catch(console.error);
 after the `callback` has been invoked. In the case of reuse of streams after
 failure, this can cause event listener leaks and swallowed errors.
 
+### `stream.pipelinify(...streams)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `streams` {Stream[]|Iterable[]|AsyncIterable[]|Function[]}
+* Returns: {stream.Duplex}
+
+Combines two or more streams into a `Duplex` stream that writes to the
+first stream and reads from the last. Each provided stream is piped into
+the next, using `stream.pipeline`. If any of the streams error then all
+are destroyed, including the outer `Duplex` stream.
+
+Because `stream.pipelinify` returns a new stream that in turn can (and
+should) be piped into other streams, it enables composition. In contrast,
+when passing streams to `stream.pipeline`, typically the first stream is
+a readable stream and the last a writable stream, forming a closed
+circuit.
+
+```js
+const { pipelinify, Transform } = require('stream');
+const removeSpaces = new Transform({
+  transform(chunk, encoding, callback) {
+    callback(null, String(chunk).replace(' '));
+  }
+});
+async function* toUpper(source) {
+  for await (const chunk of source) {
+    yield String(chunk).toUpperCase();
+  }
+}
+
+const removeSpacesAndToUpper = pipelinify(removeSpaces, toUpper);
+removeSpacesAndToUpper
+  .end('hello world')
+  .on('data', (buf) => {
+    console.log(buf); // prints 'HELLOWORLD'
```

Comment:
This actually prints `<Buffer 48 45 4c 4c 4f 55 4e 44 45 46 49 4e 45 44 57 4f 52 4c 44>`, which corresponds to `HELLOUNDEFINEDWORLD`